### PR TITLE
Feat resolve#6

### LINKED ISSUE #6: Listing movies
LET'S TO IMPLEMENT MOVIES LIST!!!

image

what can i say about it? This image says the focus of this task 😅️

### Utils
- [moviee.zip](https://github.com/joao-leal12/Movie-/files/9892732/moviee.zip) - figma layout
- [logo.zip](https://github.com/joao-leal12/Movie-/files/9892913/logo.zip)

got it

### PR DESCRIPTION
**Pull request by joao-leal12**
resolve #6

### CODE REVIEW

File: colorsDarkMode.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+DarkBG : 12190e;
```
Comment (saleszera) [inline colorsDarkMode.txt:1]: man why do you have it?

File: src/components/ContainerMovies/index.tsx [hunk 2]
```diff
@@ -1,9 +1,38 @@
...
+
+  const getMovieData = async () => {
+    const { url } = GET_MOVIES(1);
+    const { results } = await request(url);
+
+    setDataMovies(results);
+  };
+
+  useEffect(() => {
+    void getMovieData();
```
Comment (saleszera) [inline src/components/ContainerMovies/index.tsx:22]: bro, it's unnecessary to use void here

File: src/components/ContainerMovies/index.tsx [hunk 3]
```diff
@@ -1,9 +1,38 @@
...
+  }, []);
+  useEffect(() => {
+    if (ElementsFiltered !== undefined && ElementsFiltered.length > 0) {
+      setDataMovies(ElementsFiltered);
+    }
+    if (newElement.length === 0) {
+      void getMovieData();
```
Comment (saleszera) [inline src/components/ContainerMovies/index.tsx:29]: in this one too

File: src/components/Genre/index.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  const { url } = GET_GENRE();
+  const { request } = UseFetch();
+
+  useEffect(() => {
+    const GetGenreMovie = async () => {
+      const response = await request(url);
+
+      setInforGenres(response.genres);
+    };
+    void GetGenreMovie();
```
Comment (saleszera) [inline src/components/Genre/index.tsx:18]: the same here

File: src/components/GenreMoviesList/index.tsx [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  const { url } = GET_MOVIES(1);
+  interface IGenre {
+    id: number;
+    name: string;
+  }
+
+  useEffect(() => {
+    const { url } = GET_GENRE();
+
+    void request(url)
```
Comment (saleszera) [inline src/components/GenreMoviesList/index.tsx:22]: man it does not make sense cause you can filter movies by query params... But i saw the API doc and to filter by genre you need to send a POST to `/discover/movie` and the payload is weird
```cURl
curl 'https://www.themoviedb.org/discover/movie' \
  -H 'authority: www.themoviedb.org' \
  -H 'accept: text/html, */*; q=0.01' \
  -H 'accept-language: en-BR,en;q=0.9,pt-BR;q=0.8,pt;q=0.7,en-GB;q=0.6,en-US;q=0.5' \
  -H 'content-type: application/x-www-form-urlencoded; charset=UTF-8' \
  -H 'cookie: _ga=GA1.2.983548559.1666386231; tmdb.prefs=%7B%22adult%22%3Afalse%2C%22i18n_fallback_language%22%3A%22en-US%22%2C%22locale%22%3A%22pt-BR%22%2C%22country_code%22%3A%22BR%22%2C%22timezone%22%3A%22America%2FSao_Paulo%22%7D; tmdb.session=AXQloXQNfITM0bnIcQjEXuZmpWJ8tW2v3tQWDa6zogZxliE1wF7btapxnbaNTRUcF6U54rAidmCHbACIEskRbXShaTZovuaerrmR0WFg1tUFiPNnlIEn0YLddfkhuCqD5ocVsTD9L5fETmIJPIZzaP81JyHYqL6SYZlCfyp8HsDE0yZSsya6ieB9i1qNiq3aOzza8Ub2z6IKqKHlDj031X87QZWjHlH827_Inu2d9Z8u; _gid=GA1.2.963445949.1672792893; _dc_gtm_UA-2087971-10=1; _gali=with_genres' \
  -H 'dnt: 1' \
  -H 'origin: https://www.themoviedb.org' \
  -H 'referer: https://www.themoviedb.org/movie' \
  -H 'sec-ch-ua: "Not?A_Brand";v="8", "Chromium";v="108", "Google Chrome";v="108"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "Linux"' \
  -H 'sec-fetch-dest: empty' \
  -H 'sec-fetch-mode: cors' \
  -H 'sec-fetch-site: same-origin' \
  -H 'user-agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/108.0.0.0 Safari/537.36' \
  -H 'x-requested-with: XMLHttpRequest' \
  --data-raw 'air_date.gte=&air_date.lte=2023-07-04&certification=&certification_country=BR&debug=&first_air_date.gte=&first_air_date.lte=&ott_region=BR&page=1&primary_release_date.gte=&primary_release_date.lte=&region=&release_date.gte=&release_date.lte=2023-07-04&show_me=0&sort_by=popularity.desc&vote_average.gte=0&vote_average.lte=10&vote_count.gte=0&with_genres=12&with_keywords=&with_networks=&with_origin_country=&with_original_language=&with_ott_monetization_types=&with_ott_providers=&with_release_type=&with_runtime.gte=0&with_runtime.lte=400' \
  --compressed
```

well, you'll send genre ID in `with_genres` and you'll have all movies about that genre.

File: src/hooks/UseFetch.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
+import { useState, useCallback } from 'react';
+import axios from 'axios';
+
+export const UseFetch = () => {
```
Comment (saleszera) [inline src/hooks/UseFetch.ts:4]: bro you can make it better...

File: src/utils/API_KEY.ts [hunk 7]
```diff
@@ -0,0 +1 @@
+export const API_KEY: string = '9cfd40b43fdd0d0687798042ba3050eb';
```
Comment (saleszera) [inline src/utils/API_KEY.ts:1]: NOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOO
Man never do it!
API key must be saved in `.env` files

**Review by saleszera — COMMENTED** [review]
we need to make a call to talk about some things in your code

**Review by saleszera — CHANGES_REQUESTED** [review]
you forgot to use "resolve #issuenumber

File: src/hooks/UseFetch.ts (hunk 6 above)
Comment (joao-leal12) [reply src/hooks/UseFetch.ts:4]: Resolve!

**Comment by saleszera** [comment]
Man on description of your PR, you must have to write "resolve #issueId" for example "resolve #6"
About your branch name, it must have a short description of your task like "feat-card" or "feat-movie-filters"... do you understand?

**Comment by saleszera** [comment]
about [colorsDarkMode](https://github.com/joao-leal12/Movie-/blob/feat-resolve%236/colorsDarkMode.txt) why do you use `.txt` instead `.json`, i think it would better than `.txt`

**Comment by saleszera** [comment]
Man the [API_KEY](src/utils/API_KEY.ts) file you must have to DELETE!
Keep it in `.env` file

**Pull request closed without merging**